woocommerce/wc-admin
# Add performance indicators to dashboard

Fixes #1282. Closes #1281.

This PR hooks up the performance indicators on the dashboard to the new performance indicator REST API.

It also makes some more adjustments to the REST API to make this possible including report links, description/label, chart sorting, and a method of indicating which total should be a performance indicator and of what format.

A few stats off the list in #1282 are missing, because they do not exist in the reports/REST API yet. Once added, they will automatically show up here.

🌟 This also supports extensions. You can test `Follow Up Emails` via https://github.com/woocommerce/woocommerce-follow-up-emails/pull/698 with this. 🌟

### Accessibility


### Screenshots

<img width="1236" alt="screen shot 2019-01-17 at 11 44 32 am" src="https://user-images.githubusercontent.com/689165/51337205-7ee4a700-1a54-11e9-934a-60b46844b1c9.png">

<img width="299" alt="screen shot 2019-01-17 at 11 44 44 am" src="https://user-images.githubusercontent.com/689165/51337216-8310c480-1a54-11e9-90bc-e75becdef95b.png">


### Detailed test instructions:

* Run `phpunit` and make sure all tests pass.
* Run `npm test` and make sure all tests pass.
* View the dashboard, test different date ranges, and verify performance indicators show.
* Try showing/hiding indicators and testing defaults. You can clear out your usermeta (key dashboard_performance_indicator) to test defaults.
* Bonus points: Test with follow up emails.
